lib/seahorn/BvOpSem2FatMemMgr.cc

```diff
@@ -323,96 +389,109 @@ class FatMemManager : public OpSemMemManager {
   FatMemValTy MemSet(PtrTy ptr, Expr _val, unsigned len, FatMemValTy mem,
                      uint32_t align) override {
     return mkFatMem(
-        m_mem.MemSet(mkRawPtr(ptr), _val, len, mkRawMem(mem), align));
+        m_main.MemSet(mkRawPtr(ptr), _val, len, mkRawMem(mem), align),
+        m_main.MemSet(mkRawPtr(ptr), _val, len, mkSlot0Mem(mem), align),
+        m_main.MemSet(mkRawPtr(ptr), _val, len, mkSlot1Mem(mem), align));
   }
 
   /// \brief Executes symbolic memcpy with concrete length
   FatMemValTy MemCpy(PtrTy dPtr, PtrTy sPtr, unsigned len, Expr memTrsfrRead,
                      uint32_t align) override {
-    return mkFatMem(m_mem.MemCpy(mkRawPtr(dPtr), mkRawPtr(sPtr), len,
-                                 mkRawMem(memTrsfrRead), align));
+    return mkFatMem(m_main.MemCpy(mkRawPtr(dPtr), mkRawPtr(sPtr), len,
+                                  mkRawMem(memTrsfrRead), align),
+                    m_slot0.MemCpy(mkRawPtr(dPtr), mkRawPtr(sPtr), len,
+                                   mkSlot0Mem(memTrsfrRead), align),
+                    m_slot1.MemCpy(mkRawPtr(dPtr), mkRawPtr(sPtr), len,
+                                   mkSlot1Mem(memTrsfrRead), align));
   }
 
   /// \brief Executes symbolic memcpy from physical memory with concrete
   /// length
   FatMemValTy MemFill(PtrTy dPtr, char *sPtr, unsigned len, FatMemValTy mem,
                       uint32_t align = 0) override {
     return mkFatMem(
-        m_mem.MemFill(mkRawPtr(dPtr), sPtr, len, mkRawMem(mem), align));
+        m_main.MemFill(mkRawPtr(dPtr), sPtr, len, mkRawMem(mem), align),
+        m_main.MemFill(mkRawPtr(dPtr), sPtr, len, mkSlot0Mem(mem), align),
+        m_main.MemFill(mkRawPtr(dPtr), sPtr, len, mkSlot1Mem(mem), align));
   }
 
   /// \brief Executes inttoptr conversion
   FatPtrTy inttoptr(Expr intVal, const Type &intTy, const Type &ptrTy) const {
-    return mkFatPtr(m_mem.inttoptr(intVal, intTy, ptrTy));
+    return mkFatPtr(m_main.inttoptr(intVal, intTy, ptrTy),
```

Comment:
I assume that `inttoptr` sets slots to default values. Now they are set to the value of the integer. This will typecheck now, but might not in the future when slots are of different width than pointer size.

Reply:
todo

Reply:
Ack. Fixed